chainstate/merkle_root.go

```diff
@@ -0,0 +1,77 @@
+package chainstate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/BuxOrg/bux/utils"
+)
+
+type merkleRootProvider interface {
+	getName() string
+	verifyMerkleRoots(ctx context.Context, merkleProofs []string) error
+}
+
+// result of single merkle root verification by provider
+type verifyResult struct {
+	isError  bool
+	err      error
+	provider string
+}
+
+// VerifyMerkleRoots will try to verify merkle roots with all available providers
+func (c *Client) VerifyMerkleRoots(ctx context.Context, merkleRoots []string) error {
+	var wg sync.WaitGroup
+	resultsChannel := make(chan verifyResult)
+
+	for _, merkleRootPvdr := range createMerkleRootsProviders(c) {
```

Comment:
I'm sorry, but I need to say that,
From my perspective it is overcomplicated.
We don't have a requirement for using multiple "header services" to verify merkleroots. I can't even think about good use case for that.

Reply:
I agree with you but just wanted to be consistent with the structure elsewhere (for example broadcast). If you think that this is too much I can simplify this to only Pulse provider

Reply:
Removed channel and whole additional structure. Now we have simple check with Pulse